project/types.go

```diff
@@ -25,18 +32,83 @@ type UserInfo struct {
 	Value string
 }
 
+/*
+	Configuration File Definition
+ */
 type Conf struct {
-	BoltDataDir string `yaml:"bolt_data_dir"`
-	BoltDataFile string `yaml:"bolt_data_file"`
-	PostgresHost string `yaml:"pg_host"`
-	PostgresPort int `yaml:"pg_port"`
-	PostgresDatabaseName string `yaml:"pg_db_name"`
-	PostgresUser string `yaml:"pg_user"`
-	PostgresPassword string `yaml:"pg_password"`
-	PostgresSslMode string `yaml:"pg_ssl_mode"`
 	LogFile string `yaml:"log_file"`
+
+	Bolt struct {
+		DataDir string `yaml:"data_dir"`
+		DataFile string `yaml:"data_file"`
+	}
+
+	Psql struct {
+		Host string `yaml:"host"`
+		Port int `yaml:"port"`
+		DatabaseName string `yaml:"db_name"`
+		User string `yaml:"user"`
+		Password string `yaml:"password"`
+		SslMode string `yaml:"ssl_mode"`
+	}
+
+	Kube struct {
+		SourceConfigFile string `yaml:"src_config_file"`
+		DestConfigFile string `yaml:"dest_config_file"`
+	}
+}
```

Comment:
We've used [viper](https://github.com/spf1) for configuration on a bunch of other go projects.  Probably outside the scope of this PR, but something to think about.

Reply:
Cool, I'll look into that for another PR.